### PR TITLE
Added database indexes to the ObjectChange model

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #5012 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/models/change_logging.py [hunk 1]
```diff
@@ -133,6 +133,22 @@ class Meta:
                 name="extras_objectchange_rtime_idx",
                 fields=["-time"],
             ),
+            models.Index(
+                name="changed_object_double_idx",
```
Comment (timizuoebideri1) [inline nautobot/extras/models/change_logging.py:137]: The 30 character limit prevents me from using the field naming standard used above, i.e., `extras_objectchange_....`

File: changes/5012.added [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Added database indexes to the ObjectChange model:
+
+- Added index on field `user_name`
+- Added `changed_object_double_idx` index on fields: `changed_object_type`, `changed_object_id`
+- Added `related_object_double_idx` index on fields: `related_object_type`, `related_object_id`
+- Added `user_changed_object_triple_idx` index on fields: `user`, `changed_object_type`, `changed_object_id`
+- Added `name_changed_object_triple_idx` index on fields: `user_name`, `changed_object_type`, `changed_object_id`
```
Comment (glennmatthews) [inline changes/5012.added:7]: Reminder that towncrier will automatically convert each line of the change fragment to a distinct bullet list item - multi-line formatting in the change fragment will almost certainly not be formatted the way you might expect. 

Additionally, remember that the change log is primarily meant to be user-facing, so I think this is probably more detail than we need.

I'd suggest:

```suggestion
Added database indexes to the ObjectChange model to improve performance when filtering by `user_name`, `changed_object`, or `related_object`, and also by `changed_object` in combination with `user` or `user_name`.
```

File: nautobot/extras/models/change_logging.py (hunk 1 above)
Comment (glennmatthews) [reply nautobot/extras/models/change_logging.py:137]: Personally I don't think that `double` or `triple` are particularly useful terms to include in these names, though I'm not a big fan of including `extras_objectchange` in the name either. I might suggest for the new indices:

- `changed_object_idx`
- `related_object_idx`
- `user_changed_object_idx`
- `user_name_changed_object_idx`

File: nautobot/extras/models/change_logging.py [hunk 3]
```diff
@@ -133,6 +133,22 @@ class Meta:
...
+                fields=["related_object_type", "related_object_id"],
+            ),
+            models.Index(
+                name="user_changed_object_triple_idx",
+                fields=["user", "changed_object_type", "changed_object_id"],
+            ),
+            models.Index(
+                name="name_changed_object_triple_idx",
+                fields=["user_name", "changed_object_type", "changed_object_id"],
+            ),
```
Comment (glennmatthews) [inline nautobot/extras/models/change_logging.py:151]: Do we need/want a combo index on `user/related_object` and `user_name/related_object` as well? I know it wasn't called out in #5012 but it "feels" like a gap to me. @lampwins any input?

Comment (lampwins) [reply nautobot/extras/models/change_logging.py:151]: Maybe. The related objects are very rarely used and thus even less accessed. I'd be fine leaving them out for now, unless you guys really want to add them.

**Review by glennmatthews — APPROVED** [review]
LGTM

**Pull request merged**